tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/upgrade/animated/DescriptorsPage.java

```diff
@@ -100,6 +107,40 @@ public int getGridLayoutCount()
         return 2;
     }
 
+    public boolean checkDefineEncoding( File srcFile )
+    {
+        try
+        {
+            BufferedReader bReader = new BufferedReader( new StringReader( FileUtil.readContents( srcFile ) ) );
```

Comment:
there should be XML apis that you can call to get encoding instead of regex

Reply:
This, I tried to find some internal api and also checked the dom source code how to get xml definition encoding method. Their implementation use this too.

Reply:
I saw the xml specification in w3c that the encoding information shouldn't invoke by client, the encoding will be automatically parsed by any xml parser lib. This didn't directly to expose to client, the lib will be compare the hex codes in binary mode before lib start to parse xml file. I got these information from apache XmlStreamReader lib.

Reply:
In this case I think we need to get the source xml whether contain "encoding=UTF-8" string, not need to get actual encoding of xml file. Base on this, its no useful for us the internal to get encoding information.